GVRf/Framework/src/org/gearvrf/scene_objects/GVRModelSceneObject.java

```diff
@@ -0,0 +1,69 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.scene_objects;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.gearvrf.GVRContext;
+import org.gearvrf.GVRSceneObject;
+import org.gearvrf.PrettyPrint;
+import org.gearvrf.animation.GVRAnimation;
+import org.gearvrf.utility.Log;
+
+/**
+ * {@linkplain GVRSceneObject Scene object} that holds a loaded model.
+ */
+
+public class GVRModelSceneObject extends GVRSceneObject {
```

Comment:
I know this is hard to understand. However, animations loaded from a model can only affect nodes in the model. We need something like this to keep the namespace of the model. Of course, we can created animations outside the model like we are doing right now.